include/rocksdb/table.h

```diff
@@ -582,6 +582,17 @@ struct BlockBasedTableOptions {
   // Align data blocks on lesser of page size and block size
   bool block_align = false;
 
+  // Align data blocks on super block alignment. Avoid a data block split across
+  // super block boundaries.
+  bool super_block_align = false;
+
+  // Super block alignment size. Default to 512 KB. It has to be a power of 2
+  // and higher than block size.
+  size_t super_block_alignment_size = 512 * 1024;
```

Comment:
There could be use cases for non-power-of-2 sizes, e.g. encrypted files with authentication tags at fixed intervals outside the visibility of RocksDB. So for example there could be a 16 byte tag in every 64KB encryption block, so we would want super_block_alignment_size = 64 * 1024 - 16.

However, that could be added later with an API generalization not a change, so I'm ok with it with the limitation. A standard implementation using integer division could be quite slow.